compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -533,6 +539,11 @@ object TreeChecker {
         i"owner chain = ${tree.symbol.ownersIterator.toList}%, %, ctxOwners = ${ctx.outersIterator.map(_.owner).toList}%, %")
     }
 
+    override def typedTypeDef(tdef: untpd.TypeDef, sym: Symbol)(using Context): Tree = {
+      assert(isTypeDefInfo(sym.info), i"wrong type, expect a template or type bounds for ${sym.fullName}, but found: ${sym.info}")
```

Comment:
This is fine as-is, but could also be checked with an or-type type test:
```suggestion
      assert(sym.info.isInstanceOf[ClassInfo | TypeBounds], i"wrong type, expect a template or type bounds for ${sym.fullName}, but found: ${sym.info}")
```